v2/blockstore/readonly.go

```diff
@@ -404,3 +434,13 @@ func (b *ReadOnly) closeWithoutMutex() error {
 	}
 	return nil
 }
+
+// asyncErrHandlerFrom gets an error handler from ctx by checking if a value is associated with key
+// asyncErrHandlerKey. If no such value is present then it will return noopErrHandler.
+func asyncErrHandlerFrom(ctx context.Context) func(error) {
+	value := ctx.Value(asyncErrHandlerKey)
+	if eh, ok := value.(func(error)); ok {
```

Comment:
nit: check for `eh != nil` instead of `ok`, because otherwise you can't guarantee that this func never returns nil. otherwise, I can supply a nil func to WithAsyncErrorHandler and make AllKeysChan panic.

the alternative is to make AllKeysChan check `errHandler != nil` before each call, in which case you would also just `return nil` in the fallback instead of using a noop func. I'd probably go that route myself, because you also gain the ability to easily tell if an error handler is being used or not.